simply-static-deploy.php

```diff
@@ -21,4 +21,4 @@
 register_activation_hook(__FILE__, '\Grrr\SimplyStaticDeploy\Core\activate' );
 register_deactivation_hook(__FILE__, '\Grrr\SimplyStaticDeploy\Core\deactivate' );
 
-(new SimplyStaticDeploy)->init();
+(new SimplyStaticDeploy(SIMPLY_STATIC_DEPLOY_PATH))->init();
```

Comment:
I like that it's passed into the constructor, but I think that makes it less relevant to have the global constant anyway?  Maybe @HammenWS can comment on this? I think he mentioned these constants a few days ago.

If we're going down the constructor-route, I'd say `SIMPLY_STATIC_DEPLOY_VERSION` should also be passed down, because that's also used in the `Admin` class.

Reply:
Yeah, honestly, I could do without constants in our plugin internals. It's fine that a consumer would configure the plugin like that, but internally there are better choices (like parameters).
A constant is just a global variable, with all the problems that global variables carry. (except for them being changed by other parts of the system)

Reply:
I also like to have the config in injected in the constructor. Then it is the external developer's choice how to pass the configuration.

Reply:
I'd say this has nothing to with external developers right? (removing those constants also ensures that, ha).

So we're dropping this then, right?

```php
define('SIMPLY_STATIC_DEPLOY_VERSION', '0.1.0');
define('SIMPLY_STATIC_DEPLOY_URL', plugin_dir_url(__FILE__));
define('SIMPLY_STATIC_DEPLOY_PATH', plugin_dir_path(__FILE__));
````